### DISCUSSION TITLE
Loading cookies into a session

### DISCUSSION THREAD

**Discussion by Serene-Arc** [aio-libs/aiohttp#5692] [Q&A]
Hi, I've been struggling with an aspect of aiohttp for a while now and I can't seem to find any solution at all for this.

I'm converting an old project of mine from using requests to aiohttp to take advantage of the asynchronicity. However, I can't seem to find a single way to load in my cookies! It's a podcast downloader and some of the feeds are protected and need to have the user log in. This can be circumvented with cookies but I can't seem to find any way load them into the `ClientSession()`. 

I've tried everything I can think of, with the few examples I can find online. The cookies are loaded into a `MozillaCookieJar` which is part of the Python standard library but that's incompatible with `aiohttp`. Everything I've tried results in ValueErrors from tuple unpackings and other errors.

```python
    jar = MozillaCookieJar(str(cookies))
    jar.load()
    cookies = {}
    for c in jar:
        cookies[c.name] = c.value
    async with aiohttp.ClientSession(cookies=cookies) as session:
```

This doesn't work because there are RFC non-compliant cookies in the MozillaCookieJar. But if I try and filter them out, I get more `ValueErrors` from all the wretched tuple unpacking.

```python
    jar = MozillaCookieJar(str(cookies))
    jar.load()
    new_jar = aiohttp.CookieJar()
    for c in jar:
        try:
            cookie = SimpleCookie()
            cookie[c.name] = c.value
            new_jar.update_cookies(cookie)
        except http.cookies.CookieError:
            pass
    async with aiohttp.ClientSession(cookies=new_jar) as session:
        pass
```

I'm stumped and this problem is going to make me cry if I stare at it alone any more, so any help would be appreciated! I'm sure it's something simple but I really can't see it, thanks in advance :)

**Comment by Dreamsorcerer** [comment]
Maybe worth looking at some of the code that implements this:

MozillaCookieJar: https://github.com/python/cpython/blob/main/Lib/http/cookiejar.py
aiohttp passes the dict into aiohttp.CookieJar.update_cookies(): https://github.com/aio-libs/aiohttp/blob/4d58400477331704132cc9427d66699e5815ee25/aiohttp/cookiejar.py#L136

You'll also note that the type can be a BaseCookie or Morsel: https://github.com/aio-libs/aiohttp/blob/4d58400477331704132cc9427d66699e5815ee25/aiohttp/typedefs.py#L30
So, my first thought is that you could pre-encode them as one of these objects, rather than passing in str values.

I don't have any time to try it out, but hopefully that'll get you somewhere.

**Reply by Serene-Arc** [reply]
I passed in a SimpleCookie which is the simplest implementation of BaseCookie, and it successfully adds the cookie to the CookieJar, but it crashes with the tuple unpacking error when I try and use that CookieJar in the ClientSession. So it works the first time, but not the second for whatever reason.
